apps/server/src/modules/board/repo/recursive-delete.vistor.ts

```diff
@@ -0,0 +1,47 @@
+import { EntityManager } from '@mikro-orm/mongodb';
+import { Injectable } from '@nestjs/common';
+import {
+	AnyBoardDo,
+	BoardCompositeVisitorAsync,
+	BoardNode,
+	Card,
+	Column,
+	ColumnBoard,
+	TextElement,
+} from '@shared/domain';
+
+@Injectable()
+export class RecursiveDeleteVisitor implements BoardCompositeVisitorAsync {
+	constructor(private readonly em: EntityManager) {}
+
+	async visitColumnBoardAsync(columnBoard: ColumnBoard): Promise<void> {
+		this.deleteNode(columnBoard);
+		await this.visitChildrenAsync(columnBoard);
+	}
+
+	async visitColumnAsync(column: Column): Promise<void> {
+		this.deleteNode(column);
+		await this.visitChildrenAsync(column);
+	}
+
+	async visitCardAsync(card: Card): Promise<void> {
+		this.deleteNode(card);
+		await this.visitChildrenAsync(card);
+	}
+
+	async visitTextElementAsync(textElement: TextElement): Promise<void> {
```

Comment:
(-) At the moment it is a super class that know anything about the different childs. This class extends with every new card-element child type. 
(+-) The code is seperated from the effected area. Means you add a new card-element type, but you must know that you also add code on this place.